fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -1259,6 +1260,128 @@ static Any any() {
         return Any.INSTANCE;
     }
 
+    /**
+     * Special {@link Type.Record} that is undefined.
+     */
+    class AnyRecord implements Type {
+        @Nonnull
+        private final Supplier<Integer> hashCodeSupplier = Suppliers.memoize(this::computeHashCode);
+
+        @Nonnull
+        private static final AnyRecord INSTANCE = new AnyRecord();
+
+        private int computeHashCode() {
+            return Objects.hash(getTypeCode().name().hashCode(), isNullable());
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public TypeCode getTypeCode() {
+            return TypeCode.RECORD;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isNullable() {
+            return true;
+        }
+
+        @Nonnull
+        @Override
+        public AnyRecord withNullability(final boolean newIsNullable) {
+            Verify.verify(newIsNullable);
```

Comment:
this should set the attribute `isNullable` and it should not rely on it being `true`.